server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/filter/JMAPFilteringTest.java

```diff
@@ -37,95 +73,186 @@
-            Arguments.of("multiple headers", Rule.Condition.Field.TO, mimeMessageBuilder().addToRecipient(RECIPIENT_TO_1_FULL_ADDRESS).addToRecipient(RECIPIENT_TO_2_FULL_ADDRESS), RECIPIENT_TO_2_USERNAME),
-            Arguments.of("scrambled content", Rule.Condition.Field.TO, mimeMessageBuilder().addHeader("to", FRED_MARTIN_FULL_SCRAMBLED_ADDRESS), FRED_MARTIN_FULLNAME),
-            Arguments.of("folded content", Rule.Condition.Field.TO, mimeMessageBuilder().addHeader("to", SENDER_1_AND_UNFOLDED_USER_FULL_ADDRESS), UNFOLDED_USERNAME),
+    static FilteringArgumentBuilder argumentBuilder() {
+        return new FilteringArgumentBuilder();
+    }
 
-            Arguments.of("normal content", Rule.Condition.Field.CC, mimeMessageBuilder().addCcRecipient(RECIPIENT_CC_1_FULL_ADDRESS), "recipient_cc_1"),
-            Arguments.of("multiple headers", Rule.Condition.Field.CC, mimeMessageBuilder().addCcRecipient(RECIPIENT_CC_1_FULL_ADDRESS).addCcRecipient(RECIPIENT_CC_2_FULL_ADDRESS), "recipient_cc_2"),
-            Arguments.of("scrambled content", Rule.Condition.Field.CC, mimeMessageBuilder().addHeader("cc", FRED_MARTIN_FULL_SCRAMBLED_ADDRESS), FRED_MARTIN_FULLNAME),
-            Arguments.of("folded content", Rule.Condition.Field.CC, mimeMessageBuilder().addHeader("cc", SENDER_1_AND_UNFOLDED_USER_FULL_ADDRESS), UNFOLDED_USERNAME),
+    static FilteringArgumentsProvider argumentsProvider() {
+        return new FilteringArgumentsProvider();
+    }
 
-            Arguments.of("normal content", Rule.Condition.Field.SUBJECT, mimeMessageBuilder().setSubject(SCRAMBLED_SUBJECT), "subject"),
-            Arguments.of("scrambled content", Rule.Condition.Field.SUBJECT, mimeMessageBuilder().setSubject(UNSCRAMBLED_SUBJECT), "subject Frédéric MARTIN")
+    static Stream<Arguments> mailDirectiveShouldBeSetWhenContainsRuleValueParamsProvider() {
+        return Stream.concat(
+            Stream.of(FROM, TO, CC)
+                .flatMap(headerField -> argumentsProvider()
+                    .argument(argumentBuilder().description("normal content").field(headerField).header(headerField.asString(), USER_1_FULL_ADDRESS).valueToMatch(USER_1_USERNAME))
+                    .argument(argumentBuilder().description("multiple headers").field(headerField).header(headerField.asString(), USER_1_FULL_ADDRESS).from(USER_2_FULL_ADDRESS).valueToMatch(USER_1_USERNAME))
+                    .argument(argumentBuilder().description("scrambled content").field(headerField).header(headerField.asString(), FRED_MARTIN_FULL_SCRAMBLED_ADDRESS).valueToMatch(FRED_MARTIN_FULLNAME))
+                    .argument(argumentBuilder().description("folded content").field(headerField).header(headerField.asString(), USER_1_AND_UNFOLDED_USER_FULL_ADDRESS).valueToMatch(UNFOLDED_USERNAME))
+                    .argument(argumentBuilder().description("multiple spaces content").field(headerField).header(headerField.asString(), GA_BOU_ZO_MEU_FULL_ADDRESS).valueToMatch(BOU))
```

Comment:
You can `.map(field -> argumentBuilder().field(headerField))` to avoid repeating yourself everywhere